webapp/src/main/java/com/box/l10n/mojito/service/tm/search/TextUnitSearcher.java

```diff
@@ -179,6 +179,14 @@ NativeCriteria getCriteriaForSearch(TextUnitSearcherParameters searchParameters)
               "tm_text_unit_pending_mt", "tmtupmt", "tmtupmt.tm_text_unit_id", "tu.id"));
     }
 
+    if (searchParameters.getStatusFilter() != null
+        && searchParameters.getStatusFilter().equals(StatusFilter.MT_TRANSLATED)) {
+      // Pushing AI translations, need to retrieve the uploadedFileUri from the ThirdPartyTextUnit
```

Comment:
I'm reading the comment but still not sure I understand why this is needed here - in the TextUnitSearcher.

Couldn't we retrieve that attribute just where required, outside of the searcher, if uploadedFileUri is in no way used in the base search? (to avoid slowing down other queries that don't need it?)

Reply:
My thinking around this was to retrieve the uploadedFileUri information at the same time as we retrieve the TextUnitDTOs in the TextUnitSearcher, so that there's a single DB retrieval call to retrieve all the information we require for the push of AI translations. 

I restrict the projection to only occur when the MT_TRANSLATED status filter is used to prevent it impacting other queries, I can alternatively update that to a different boolean (`retrieveFileUploadUri`) so that it's more specific around how it's affecting the retrieval if that's preferable?

I can also do that retrieval as a separate DB call in the push AI logic after we've retrieved the TextUnitDTOs if you feel that would be cleaner?

Reply:
When we would want to see just MT_TRANSLATED from the UI for example, we wouldn't care at all about the fileUploadUri - and taking the corresponding perf hit from that.

Either of the options you mentioned: the bool or the separate query would solve that problem.

Reply:
I'll update to use a specific boolean so, as the other approach requires a further DB call to map a TU with it's uploaded file uri.